### PR TITLE
Add binary version to status

### PR DESCRIPTION
**Pull request by Taztingo**
## Description

Related: https://github.com/provenance-io/provenance/issues/1578

Update the status command to return the binary version. A new field `binary_version` is present under `NodeInfo`.

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/main/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/main/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: client/rpc.go [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+	if err != nil {
+		return -1, err
+	}
+
+	height := status.SyncInfo.LatestBlockHeight
+	return height, nil
+}
+
+// Validator output
+type ValidatorOutput struct {
```
Comment (nullpointer0x00) [inline client/rpc.go:31]: Same as last comment.

File: client/rpc.go [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+	Address          sdk.ConsAddress    `json:"address"`
+	PubKey           cryptotypes.PubKey `json:"pub_key"`
+	ProposerPriority int64              `json:"proposer_priority"`
+	VotingPower      int64              `json:"voting_power"`
+}
+
+// Validators at a certain height output in bech32 format
+type ResultValidatorsOutput struct {
```
Comment (nullpointer0x00) [inline client/rpc.go:39]: Same as last comment.

File: client/rpc.go [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+	"strings"
+
+	cryptocodec "github.com/cosmos/cosmos-sdk/crypto/codec"
+	cryptotypes "github.com/cosmos/cosmos-sdk/crypto/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	tmtypes "github.com/tendermint/tendermint/types"
+)
+
+// get the current blockchain height
+func GetChainHeight(clientCtx Context) (int64, error) {
```
[the remark this answers is not in the document]
Comment (Taztingo) [reply client/rpc.go:15]: These aren't new functions, and they were just moved into rpc.go to remove the circular dependency. I kept the comment the same just to keep it consistent with what they had before.

**Review by nullpointer0x00 — APPROVED** [review]

**Review by SpicyLemon — APPROVED** [review]
I'd be nice if we could also get it in the `/status` rpc endpoint (e.g. `https://rpc.provenance.io:443/status`), but it just doesn't seem possible without forking cometbft/tendermint.

I did find the `https://rpc.provenance.io:443/sync_info` that we manually add to the rpc routes, and it has the binary version in it. So, between the `provenanced status` command change in this PR, and that other endpoint, we've got what we need.

**Comment by SpicyLemon** [comment]
I ran the following `make` targets locally, and they completed without issue: `test` `test-sim-nondeterminism` `test-sim-import-export` `test-sim-after-import` `test-sim-multi-seed-short`.

**Pull request merged**